src/main/java/sirius/pasta/noodle/macros/SvgResourceMacro.java

```diff
@@ -0,0 +1,149 @@
+        }
+        if (args.size() == 2 && !CompilationContext.isAssignableTo(args.get(1), String.class)) {
+            throw new IllegalArgumentException("If given, the second argument needs to be a String.");
+        }
+        if (args.size() > 2) {
+            throw new IllegalArgumentException("Expected one or two String arguments.");
+        }
+    }
+
+    @Override
+    public void verify(CompilationContext context, Position position, List<Node> args) {
+        super.verify(context, position, args);
+
+        if (args.get(0).isConstant()) {
+            String resourceName = String.valueOf(args.get(0).getConstantValue());
+            if (resources.resolve(resourceName).isEmpty()) {
+                context.warning(position, "Unknown resource: %s", resourceName);
+            }
+        }
+
+        if (args.size() > 1 && args.get(1).isConstant()) {
+            try {
+                new Color(String.valueOf(args.get(1).getConstantValue()));
+            } catch (IllegalArgumentException exception) {
+                context.warning(position, exception.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public boolean isConstant(CompilationContext context, List<Node> args) {
+        return args.get(0).isConstant() && (args.size() == 1 || args.get(1).isConstant());
+    }
+
+    @Override
+    public String getDescription() {
+        return "Creates a UTF-8 representation of the given SVG resource, possibly changed to the given tint color, to be included in IMG tags or CSS files";
+    }
+
+    @Override
+    public Object invoke(Environment environment, Object[] args) {
+        String path = (String) args[0];
+        String color = args.length > 1 ? (String) args[1] : null;
+        return encodeResource(path, color);
+    }
+
+    @Nonnull
+    private String encodeResource(String path, String color) {
+        if (!path.startsWith("/assets/")) {
+            throw new IllegalArgumentException("Only assets can be inlined for security reasons.");
+        }
+
+        // parse XML, clean whitespace and XML declaration, and synthesize a string representation
+        Document document = InlineSvgMacro.parseSvgDocument(resources.resolve(path)
+                                                                     .orElseThrow(() -> new IllegalArgumentException(
+                                                                             "Unknown resource: " + path)));
+        Element root = InlineSvgMacro.cleanIndentationAndNewlines(document.getDocumentElement());
+        String svgCode = InlineSvgMacro.stringifyElement(root, false);
+
+        // optionally replace black with given tint color
+        if (Strings.isFilled(color)) {
+            try {
+                new Color(color);
+            } catch (IllegalArgumentException exception) {
+                throw new IllegalArgumentException("Unknown color: " + color + " (Hex-color #rrggbb expected.)");
+            }
```

Comment:
extract method "verifyValidColorValue" or the like

Reply:
We can not reuse such a method as the catch clauses are different